File: Samples/VB/14-ContextualTabs/Form1.vb
Imports RibbonLib
Imports RibbonLib.Controls
Imports RibbonLib.Controls.Events
Imports RibbonLib.Interop

Namespace _14_ContextualTabs
	Public Enum RibbonMarkupCommands As UInteger
		cmdTabMain = 1001
		cmdGroupMain = 1002
		cmdTabGroupTableTools = 1003
		cmdTabDesign = 1004
		cmdTabLayout = 1005
		cmdGroupDesign = 1006
		cmdGroupLayout = 1007
		cmdButtonSelect = 1008
		cmdButtonUnselect = 1009
		cmdButtonDesign1 = 1010
		cmdButtonDesign2 = 1011
		cmdButtonDesign3 = 1012
		cmdButtonLayout1 = 1013
		cmdButtonLayout2 = 1014
	End Enum

	Partial Public Class Form1
		Inherits Form
		Private _tabMain As RibbonTab
		Private _groupMain As RibbonGroup
		Private _tabGroupTableTools As RibbonTabGroup
		Private _tabDesign As RibbonTab
		Private _tabLayout As RibbonTab
		Private _groupDesign As RibbonGroup
		Private _groupLayout As RibbonGroup
		Private _buttonSelect As RibbonButton
		Private _buttonUnselect As RibbonButton
		Private _buttonDesign1 As RibbonButton
		Private _buttonDesign2 As RibbonButton
		Private _buttonDesign3 As RibbonButton
		Private _buttonLayout1 As RibbonButton
		Private _buttonLayout2 As RibbonButton

		Public Sub New()
			InitializeComponent()

			_tabMain = New RibbonTab(_ribbon, CUInt(RibbonMarkupCommands.cmdTabMain))
			_groupMain = New RibbonGroup(_ribbon, CUInt(RibbonMarkupCommands.cmdGroupMain))
			_tabGroupTableTools = New RibbonTabGroup(_ribbon, CUInt(RibbonMarkupCommands.cmdTabGroupTableTools))
			_tabDesign = New RibbonTab(_ribbon, CUInt(RibbonMarkupCommands.cmdTabDesign))
			_tabLayout = New RibbonTab(_ribbon, CUInt(RibbonMarkupCommands.cmdTabLayout))
			_groupDesign = New RibbonGroup(_ribbon, CUInt(RibbonMarkupCommands.cmdGroupDesign))
			_groupLayout = New RibbonGroup(_ribbon, CUInt(RibbonMarkupCommands.cmdGroupLayout))
			_buttonSelect = New RibbonButton(_ribbon, CUInt(RibbonMarkupCommands.cmdButtonSelect))
			_buttonUnselect = New RibbonButton(_ribbon, CUInt(RibbonMarkupCommands.cmdButtonUnselect))
			_buttonDesign1 = New RibbonButton(_ribbon, CUInt(RibbonMarkupCommands.cmdButtonDesign1))
			_buttonDesign2 = New RibbonButton(_ribbon, CUInt(RibbonMarkupCommands.cmdButtonDesign2))
			_buttonDesign3 = New RibbonButton(_ribbon, CUInt(RibbonMarkupCommands.cmdButtonDesign3))
			_buttonLayout1 = New RibbonButton(_ribbon, CUInt(RibbonMarkupCommands.cmdButtonLayout1))
			_buttonLayout2 = New RibbonButton(_ribbon, CUInt(RibbonMarkupCommands.cmdButtonLayout2))

			AddHandler _buttonSelect.ExecuteEvent, AddressOf _buttonSelect_ExecuteEvent
			AddHandler _buttonUnselect.ExecuteEvent, AddressOf _buttonUnselect_ExecuteEvent
		End Sub

		Private Sub _buttonSelect_ExecuteEvent(ByVal sender As Object, ByVal e As ExecuteEventArgs)
			_tabGroupTableTools.ContextAvailable = ContextAvailability.Active
		End Sub

		Private Sub _buttonUnselect_ExecuteEvent(ByVal sender As Object, ByVal e As ExecuteEventArgs)
			_tabGroupTableTools.ContextAvailable = ContextAvailability.NotAvailable
		End Sub
	End Class
End Namespace

File: Samples/CS/14-ContextualTabs/Form1.cs
using System;
using System.Windows.Forms;
using RibbonLib;
using RibbonLib.Controls;
using RibbonLib.Controls.Events;
using RibbonLib.Interop;

namespace _14_ContextualTabs
{
    public enum RibbonMarkupCommands : uint
    {
        cmdTabMain = 1001,
        cmdGroupMain = 1002,
        cmdTabGroupTableTools = 1003,
        cmdTabDesign = 1004,
        cmdTabLayout = 1005,
        cmdGroupDesign = 1006,
        cmdGroupLayout = 1007,
        cmdButtonSelect = 1008,
        cmdButtonUnselect = 1009,
        cmdButtonDesign1 = 1010,
        cmdButtonDesign2 = 1011,
        cmdButtonDesign3 = 1012,
        cmdButtonLayout1 = 1013,
        cmdButtonLayout2 = 1014,
    }

    public partial class Form1 : Form
    {
        private RibbonTab _tabMain;
        private RibbonGroup _groupMain;
        private RibbonTabGroup _tabGroupTableTools;
        private RibbonTab _tabDesign;
        private RibbonTab _tabLayout;
        private RibbonGroup _groupDesign;
        private RibbonGroup _groupLayout;
        private RibbonButton _buttonSelect;
        private RibbonButton _buttonUnselect;
        private RibbonButton _buttonDesign1;
        private RibbonButton _buttonDesign2;
        private RibbonButton _buttonDesign3;
        private RibbonButton _buttonLayout1;
        private RibbonButton _buttonLayout2;

        public Form1()
        {
            InitializeComponent();

            _tabMain = new RibbonTab(_ribbon, (uint)RibbonMarkupCommands.cmdTabMain);
            _groupMain = new RibbonGroup(_ribbon, (uint)RibbonMarkupCommands.cmdGroupMain);
            _tabGroupTableTools = new RibbonTabGroup(_ribbon, (uint)RibbonMarkupCommands.cmdTabGroupTableTools);
            _tabDesign = new RibbonTab(_ribbon, (uint)RibbonMarkupCommands.cmdTabDesign);
            _tabLayout = new RibbonTab(_ribbon, (uint)RibbonMarkupCommands.cmdTabLayout);
            _groupDesign = new RibbonGroup(_ribbon, (uint)RibbonMarkupCommands.cmdGroupDesign);
            _groupLayout = new RibbonGroup(_ribbon, (uint)RibbonMarkupCommands.cmdGroupLayout);
            _buttonSelect = new RibbonButton(_ribbon, (uint)RibbonMarkupCommands.cmdButtonSelect);
            _buttonUnselect = new RibbonButton(_ribbon, (uint)RibbonMarkupCommands.cmdButtonUnselect);
            _buttonDesign1 = new RibbonButton(_ribbon, (uint)RibbonMarkupCommands.cmdButtonDesign1);
            _buttonDesign2 = new RibbonButton(_ribbon, (uint)RibbonMarkupCommands.cmdButtonDesign2);
            _buttonDesign3 = new RibbonButton(_ribbon, (uint)RibbonMarkupCommands.cmdButtonDesign3);
            _buttonLayout1 = new RibbonButton(_ribbon, (uint)RibbonMarkupCommands.cmdButtonLayout1);
            _buttonLayout2 = new RibbonButton(_ribbon, (uint)RibbonMarkupCommands.cmdButtonLayout2);

            _buttonSelect.ExecuteEvent += new EventHandler<ExecuteEventArgs>(_buttonSelect_ExecuteEvent);
            _buttonUnselect.ExecuteEvent += new EventHandler<ExecuteEventArgs>(_buttonUnselect_ExecuteEvent);
        }

        void _buttonSelect_ExecuteEvent(object sender, ExecuteEventArgs e)
        {
            _tabGroupTableTools.ContextAvailable = ContextAvailability.Active;
        }

        void _buttonUnselect_ExecuteEvent(object sender, ExecuteEventArgs e)
        {
            _tabGroupTableTools.ContextAvailable = ContextAvailability.NotAvailable;
        }
    }
}
